src/expression.c

```diff
@@ -10015,12 +10086,12 @@ Expression *SliceExp::semantic(Scope *sc)
         }
         else if (t1b->ty == Tpointer)
         {
-            this->upperIsInBounds = true;
+            this->upperIsInBounds = true; // no boundscheck for x.ptr[_ .. _]
         }
         else
             assert(0);
 
-        this->lowerIsLessThanUpper = (lwrRange.imax <= uprRange.imin);
+        this->lowerIsLessThanUpper = this->lowerIsLessThanUpper || (lwrRange.imax <= uprRange.imin);
```

Comment:
> this does bitwise logic. But for zeros and ones I guess that's ok right?

Bitwise logic on a bool is boolean logic.  It makes sense if you don't think about it too much.